pkg/storaged/utils.js

```diff
@@ -843,7 +865,15 @@
             });
         }
 
-        if (fsys && fsys.MountPoints.length > 0) {
+        // HACK: This works for unmounting a btrfs subvolume, however on trying
+        // to mount a second btrfs volume this will try to unmount the block
+        // device where btrfs is on. So we just check if there is a mount_point.
+        // Note that we don't handle any of the overmounting stuff now.
+        if (btrfs_volume && subvol) {
+            if (subvol.mount_point) {
+                enter_unmount(block, subvol.mount_point, true);
+            }
+        } else if (fsys && fsys.MountPoints.length > 0) {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19690-20231207-124613-a587c975-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)